src/Dhall/TypeCheck.hs

```diff
@@ -425,17 +439,22 @@ typeWithA tpa = loop
                           (Pi "nothing" "optional" "optional") )
     loop ctx e@(Record    kts   ) = do
         let process (k, t) = do
-                s <- fmap Dhall.Core.normalize (loop ctx t)
+                s <- fmap normalizeTy (loop ctx t)
                 case s of
                     Const Type -> return ()
                     Const Kind -> return ()
-                    _          -> Left (TypeError ctx e (InvalidFieldType k t))
+                    other -> loop ctx other >>= \x -> case x of
+                    -- This codepath doesn't come into play for vanilla Dhall,
```

Comment:
This code path does affect vanilla Dhall because it would make the following record type legal:

```haskell
{ x : List }
```

... where previously it was forbidden

Also, it seems odd that this extension is only done for record types and not elsewhere

Reply:
Hmm that's true..

* I could make it this instead to help ensure vanilla Dhall isn't affected:
```haskell
other@(Embed _) -> loop ctx other -- etc
```
* Alternatively, this can remain and Vanilla Dhall can be affected slightly, but this newly-valid record type has no inhabitants anyways?

> Also, it seems odd that this extension is only done for record types and not elsewhere

Yeah I suppose it would make sense for Union types to have this extension as well - is that what you meant?

Reply:
Even if it has no inhabitants it should be rejected so that users get better error messages and don't get led astray

What I meant was that (A) it's not clear what this change has to do with this feature of customizing judgmental equality and (B) it's not clear why this special check is only done here instead of everywhere that the type-checker checks for a `Type` or `Kind`